python/lsst/cp/pipe/ptc.py

```diff
@@ -365,10 +376,102 @@ def runDataRef(self, dataRef, visitPairs):
         self.log.info(f"Writing PTC and NL data to {dataRef.getUri(write=True)}")
         dataRef.put(dataset, datasetType="photonTransferCurveDataset")
 
+        self.log.info(f"Writing linearizers")
+
+        detName = detector.getName()
+        now = datetime.datetime.utcnow()
+        butler = dataRef.getButler()
+
+        for linPair in [("LOOKUPTABLE", 'linearizeLut'), ("LINEARIZEPOLYNOMIAL", 'linearizePolynomial'),
+                        ("LINEARIZESQUARED", 'linearizeSquared')]:
+
+            linType = linPair[0]
+            dataType = linPair[1]
+            if linType == "LOOKUPTABLE":
+                tableArray = lookupTableArray
+            else:
+                tableArray = None
+
+            linearizer = self.buildLinearizerObject(dataset, detName, detNum, detector,
+                                                    instruName=self.config.instrumentName,
+                                                    linearizerType=linType, tableArray=tableArray,
+                                                    log=self.log)
+            butler.put(linearizer.toDict(), datasetType=dataType, dataId={'detector': detNum,
```

Comment:
I tried getting rid of "toDict" and it did not work properly.  It seems that gen2 butler expects a dict.

Reply:
You would need to teach it to do the right thing like we taught butler to understand that Defects can be put.

Reply:
I see that cp_pipe doesn't use the butler for defects but does try to use the butler for ptc. I think that explains why Butler.put has not been set up for Defects. It should be simple for Linearizer to have the dataset type added to gen2 butler since it emulates a normal Catalog by supporting writeFits.

Reply:
So the linearizer should be treated like a fits catalog and written as a fits catalog? Should not they be human-readable?

Reply:
You are putting them into a butler so the format does not have to be the same as the format you use in obs_lsst_data.   If you have been copying the files out of the butler manually then that would make a difference (otherwise you'd need a quick readFits/writeText copy-like command).  I don't know how easy it is to add a new formatter in gen2 that would call writeText directly so maybe it's not going to be possible to do this in gen2 properly.

Reply:
At this moment I don't know either how easy what you describe would be. Could this be left for a future ticket, if necessary?